internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -835,233 +841,380 @@ func resourceCosmosDbAccountCreate(d *pluginsdk.ResourceData, meta interface{})
+			log.Printf("[INFO] Updating AzureRM Cosmos DB Account: Updating 'Locations'")
+			// add any new/renamed locations
+			accountProps.Locations = pointer.To(configLocations)
+
+			// Update the database locations...
+			err = resourceCosmosDbAccountApiCreateOrUpdate(client, ctx, id.ResourceGroup, id.Name, account, d)
+			if err != nil {
+				return fmt.Errorf("updating %q `locations`: %+v", id, err)
+			}
+		} else {
+			log.Printf("[INFO] [SKIP] AzureRM Cosmos DB Account: Updating 'Locations' [NO CHANGE]")
 		}
-	}
 
-	if v, ok := d.GetOk("backup"); ok {
-		policy, err := expandCosmosdbAccountBackup(v.([]interface{}), d.HasChange("backup.0.type"), createMode)
+		// Update Identity and Default Identity...
+		identityChanged := false
+		expandedIdentity, err := expandAccountIdentity(d.Get("identity").([]interface{}))
 		if err != nil {
-			return fmt.Errorf("expanding `backup`: %+v", err)
+			return fmt.Errorf("expanding `identity`: %+v", err)
 		}
-		account.DatabaseAccountCreateUpdateProperties.BackupPolicy = policy
-	} else if createMode != "" {
-		return fmt.Errorf("`create_mode` only works when `backup.type` is `Continuous`")
-	}
 
-	if err = resourceCosmosDbAccountApiUpsert(client, ctx, id.ResourceGroup, id.Name, account, d); err != nil {
-		return fmt.Errorf("updating %s: %+v", id, err)
-	}
+		if d.HasChange("identity") {
+			identityChanged = true
 
-	// Update the property independently after the initial upsert as no other properties may change at the same time.
-	account.DatabaseAccountCreateUpdateProperties.EnableMultipleWriteLocations = utils.Bool(enableMultipleWriteLocations)
-	if *resp.EnableMultipleWriteLocations != enableMultipleWriteLocations {
-		if err = resourceCosmosDbAccountApiUpsert(client, ctx, id.ResourceGroup, id.Name, account, d); err != nil {
-			return fmt.Errorf("updating %s EnableMultipleWriteLocations: %+v", id, err)
-		}
-	}
+			// Looks like you have to always remove all the identities first before you can
+			// reassign/modify them, else it will append any new/changed identities
+			// resulting in a diff...
+			log.Printf("[INFO] Updating AzureRM Cosmos DB Account: Setting 'Identity' to 'None'")
+
+			// can't set this back to account, because that will hit the bug...
+			identity := documentdb.DatabaseAccountUpdateParameters{
+				Identity: pointer.To(documentdb.ManagedServiceIdentity{
+					Type: documentdb.ResourceIdentityTypeNone,
+				}),
+			}
 
-	// determine if any locations have been renamed/priority reordered and remove them
-	removedOne := false
-	for _, l := range newLocations {
-		if ol, ok := oldLocationsMap[*l.LocationName]; ok {
-			if *l.FailoverPriority != *ol.FailoverPriority {
-				if *l.FailoverPriority == 0 {
-					return fmt.Errorf("cannot change the failover priority of %s location %s to %d", id, *l.LocationName, *l.FailoverPriority)
+			// Update the database 'Identity' to 'None'...
+			err = resourceCosmosDbAccountApiUpdate(client, ctx, id.ResourceGroup, id.Name, identity, d)
```

Comment:
I have a feeling this may not work when a Customer Managed Key is present, is that covered by an acctest?

Reply:
It is now... What the test does is it starts with a `SystemAssigned` identity, moves to a `SystemAssigned, UserAssigned`, then to just `UserAssigned` and finally back to just `SystemAssigned` again at the end of the test case.

**Note:** The CMK has perms assigned to the `CosmosDB` and the `UserAssigned` Identity.

<img width="446" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/20408400/4a1021d0-9ad9-4746-aaf1-8ba45d99c751">